_posts/2022-12-06-integrating-auth0-with-react.md

```diff
@@ -0,0 +1,310 @@
+```
+
+Here we will once again utilise the `useAuth0` hook this time with the `logout` method, Pretty straight forward right?
+
+The `logout` method will direct your users to your Auth0 logout endpoint, and once completed, will redirect them to your root page (i.e. `http://localhost:3000`).
+
+It can also be configured with an optional parameter `returnTo` to direct them to another part of your application if you would like.
+```ts
+logout({ returnTo: 'http://localhost:3000/logout' })
+```
+
+**Be aware, this returnTo parameter MUST be included in your Allowed Logout URLs that we configured earlier**
+
+```tsx
+// /src/App.tsx
+...
+import LoginButton from './components/LoginButton';
+import LogoutButton from './components/LogoutButton';
+
+function App() {
+  return (
+    <div className="App">
+      <header className="App-header">
+        ...
+        <a
+          ...
+        >
+          Learn React
+        </a>
+        <LoginButton />
+        <LogoutButton />
+      </header>
+    </div>
+  );
+}
+...
+```
+
+![Logout Button](/img/blog/integrating-auth0-with-react/logout.png){:width="275px" height="190px" loading="lazy"}
+
+Awesome, we now have User Signup, Login, and Logout!
+
+## User information and Profile
+
+Now that our users can login, How do we go about getting their information such as Name, Email address, Profile Picture etc?
+
+Alright, Let's create a new component that will retrieve that for us.
+
+```tsx
+// /src/components/Profile.tsx
+
+import React, { FC } from 'react';
+import { useAuth0 } from "@auth0/auth0-react";
+
+const Profile: FC = () => {
+    const { user, isAuthenticated, isLoading } = useAuth0();
+
+    if (isLoading) {
+        return <div>Loading...</div>
+    }
+
+    return (
+        <div>
+            {isAuthenticated && user && (
+                <>
+                    <img src={user.picture} alt={user.name} />
+                    <h2>{user.name}</h2>
+                    <p>{user.email}</p>
+                </>
+            )}
+        </div>
+    );
+}
+
+export default Profile;
+```
+
+Once again, the `useAuth0` hook showing its strengths! We can get the user object, check if they are authenticated, and that everything has been fully loaded before displaying the user profile.
+
+Let's add that to our homepage
+
+```tsx
+// /src/App.tsx
+...
+import LoginButton from './components/LoginButton';
+import LogoutButton from './components/LogoutButton';
+import Profile from './components/Profile';
+
+function App() {
+  return (
+    <div className="App">
+      <header className="App-header">
+        <LoginButton />
+        <LogoutButton />
+        <Profile />
+      </header>
+    </div>
+  );
+}
+
+export default App;
+```
+
+![Profile](/img/blog/integrating-auth0-with-react/profile.png){:width="580px" height="387px" loading="lazy"}
+
+There we go! A fully integrated user Authenticated system and an amazing launch point for all of your Authentication and Authorization needs.
+
+## Conclusion
```

Comment:
A question I have is where do these records get stored (I'm assuming Auth0), and what security precautions do they undertake to keep things secure? Might'n be necessary for this blog, I'm just curious :)

Reply:
Yeah, Auth0 managed database, However they do have the ability to use your own self managed database if you want!

You can also use the same database accross multiple clients

Security wise, They have pretty solid [compliance](https://auth0.com/security) and honestly, I feel like it's even easier to understand how secure they are vs the Shared Responsibility model for Cognito. it's not hidden in whitepapers and random docs across the website